modules/sdk-coin-dot/src/lib/batchTransactionBuilder.ts

```diff
@@ -69,18 +113,33 @@ export class BatchTransactionBuilder extends TransactionBuilder {
   /** @inheritdoc */
   protected fromImplementation(rawTransaction: string): Transaction {
     const tx = super.fromImplementation(rawTransaction);
-    if (this._method?.name === MethodNames.Batch) {
+    if (this._method?.name === MethodNames.Batch || this._method?.name === MethodNames.BatchAll) {
+      if (this._method?.name === MethodNames.BatchAll) {
```

Comment:
for my own understanding, what's the difference between Batch and BatchAll here?

Reply:
> [utility.batchAll()](https://github.com/paritytech/substrate/blob/ded44948e2d5a398abcb4e342b0513cb690961bb/frame/utility/src/lib.rs#L284) does not commit if one of the calls in the batch fails.
> 
> The whole transaction will rollback and fail if any of the calls failed.
> 
> [utility.batch()](https://github.com/paritytech/substrate/blob/ded44948e2d5a398abcb4e342b0513cb690961bb/frame/utility/src/lib.rs#L177) commits each successful call regardless if a call fails.
> 
> If a call failed and the batch was interrupted, then the BatchInterrupted event is deposited, along with the number of successful calls made and the error of the failed call. If all were successful, then the BatchCompleted event is deposited.

Taken from https://substrate.stackexchange.com/questions/2084/what-is-the-difference-between-the-extrinsics-utility-batch-and-utility-batcha